src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,127 @@
+                String[] splited = resource.substring(6).split("/");
+                Event newEvent = new Event(splited[0], splited[1].replace(" ", ""));
+                result = newEvent;
+            } catch (Exception e) {
+                throw new InvalidParameterException();
+            }
+        } else {
+            throw new InvalidParameterException();
+        }
+        return result;
+    }
+
+    /**
+     * Parses the done order from user.
+     * Gives the index of target task.
+     *
+     * @param resource User input string representing done order.
+     * @param taskList TaskList object of the program.
+     * @return Index of target task in taskList.
+     * @throws IndexOutOfBoundsException If user inputs an index not in task list.
+     * @throws Exception                 If user inputs done order in wrong format.
+     */
+    public int parseDoneOrder(String resource, TaskList taskList) throws IndexOutOfBoundsException, Exception {
+        if (resource.length() <= 4) {
+            throw new Exception();
+        }
+        String numberString = resource.substring(5).replace(" ", "");
+        try {
+            int doneNo = Integer.parseInt(numberString);
+            if (doneNo <= 0 || doneNo - 1 >= taskList.taskCount()) {
+                throw new IndexOutOfBoundsException();
+            } else {
+                return doneNo - 1;
+            }
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Parses the delete order from user.
+     * Gives the index of target task.
+     *
+     * @param resource User input string representing delete order.
+     * @param taskList TaskList object of the program.
+     * @return Index of deleting task in taskList.
+     * @throws IndexOutOfBoundsException If user inputs an index not in task list.
+     * @throws Exception                 If user inputs deleting order in wrong format.
+     */
+    public int parseDeleteOrder(String resource, TaskList taskList) throws IndexOutOfBoundsException, Exception {
+        if (resource.length() <= 6) {
+            throw new Exception();
+        }
+        String numberString = resource.substring(7).replace(" ", "");
+        try {
+            int deletingNumber = Integer.parseInt(numberString);
+            if (deletingNumber <= 0 || deletingNumber - 1 >= taskList.taskCount()) {
+                throw new IndexOutOfBoundsException();
+            } else {
+                return deletingNumber - 1;
+            }
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Parses the find Order from user.
+     * Gives the keyword.
+     *
+     * @param resource User input string representing find order.
+     * @return Keyword user is searching for.
+     * @throws InvalidParameterException If there is no keyword followed.
+     */
+    public String parseFindOrder(String resource) throws InvalidParameterException {
+        if (resource.length() <= 4) {
```

Comment:
Perhaps consider using a final int to store the value 4 as a constant with a comprehensible variable name, to avoid the use of "magic numbers".